clippy_lints/src/booleans.rs

```diff
@@ -406,6 +437,7 @@ impl<'a, 'tcx> Visitor<'tcx> for NonminimalBoolVisitor<'a, 'tcx> {
         match e.node {
             ExprBinary(binop, _, _) if binop.node == BiOr || binop.node == BiAnd => self.bool_expr(e),
             ExprUnary(UnNot, ref inner) => if self.cx.tables.node_types()[inner.hir_id].is_bool() {
+                self.handle_method_call_in_not(e, inner);
```

Comment:
I would prefer to include this as part of https://github.com/rust-lang-nursery/rust-clippy/pull/2216/files#diff-2874c632e14921250b07b4805c090f19R181 so things like `!(foo.is_some() && !bar)` gets optimized to `foo.is_none() || bar`

Reply:
I gave this a short try without much success. I'm not sure to understand the whole architecture correctly but wouldn't that require 'expr.simplify()' (in the bool_expr function) to also handle these negations ?

Reply:
Hmmm since I wrote the code, I'll see if I can figure it out. Looks certainly nontrivial

Reply:
Ok, let me know if there is anything I can do to help.

Reply:
So. it looks like it's not necessary to modify the `simplify` method. Or I haven't been able to create horrible enough test cases that trigger said issue.